### PR TITLE
feat(BUX-290): merkleroots issue

### PR DESCRIPTION
**Pull request by kuba-4chain**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/pulse/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/pulse/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [x] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: internal/tests/testrepository/header_testrepository.go [hunk 1]
```diff
@@ -190,23 +190,43 @@ func (r *HeaderTestRepository) GetChainBetweenTwoHashes(low string, high string)
...
 func (r *HeaderTestRepository) GetMerkleRootsConfirmations(
-	merkleroots []string,
+	request []domains.MerkleRootConfirmationRequestItem,
+	maxBlockHeightExcess int,
 ) ([]*domains.MerkleRootConfirmation, error) {
 	mrcfs := make([]*domains.MerkleRootConfirmation, 0)
 
-	for _, mr := range merkleroots {
-		hash := ""
+	var topHeight int32 = 0
```
Comment (arkadiuszos4chain) [inline internal/tests/testrepository/header_testrepository.go:198]: Shouldn't we use init style like `topHeight:= int32(0)` ?

File: config/flags.go [hunk 2]
```diff
@@ -164,6 +170,11 @@ func (fs *PulseFlagSet) bindWebsocketFlags() {
 	viper.BindPFlag(EnvWebsocketHistoryTtl, fs.Lookup(historyTTLFlag))
 }
 
+//nolint:all
```
Comment (pawellewandowski98) [inline config/flags.go:173]: why nolint comment is required here?

File: repository/dto/headers.go [hunk 3]
```diff
@@ -94,30 +94,48 @@ func ToDbBlockHeader(bh domains.BlockHeader) DbBlockHeader {
...
-		Confirmed:  dbMerkleConfm.Confirmed,
+func (dbMerkleConfm *DbMerkleRootConfirmation) ToMerkleRootConfirmation(
+	maxBlockHeightExcess int,
+) *domains.MerkleRootConfirmation {
+	var confmState domains.MerkleRootConfirmationState
+
+	if dbMerkleConfm.Hash.Valid {
+		confmState = domains.Confirmed
+	} else if dbMerkleConfm.BlockHeight > dbMerkleConfm.TipHeight &&
+		dbMerkleConfm.BlockHeight-dbMerkleConfm.TipHeight < int32(maxBlockHeightExcess+1) {
```
Comment (arkadiuszos4chain) [inline repository/dto/headers.go:113]: Please replace`<` with `<=` and remove `+1`

File: config/config.json [hunk 4]
```diff
@@ -39,6 +39,9 @@
     "db": {
 
     },
+	"merkleroot": {
```
[the remark this answers is not in the document]
Comment (kuba-4chain) [reply config/config.json]: Done

File: internal/tests/testrepository/header_testrepository.go (hunk 1 above)
Comment (kuba-4chain) [reply internal/tests/testrepository/header_testrepository.go]: True, done

File: repository/dto/headers.go (hunk 3 above)
Comment (kuba-4chain) [reply repository/dto/headers.go]: Done

File: config/flags.go (hunk 2 above)
Comment (kuba-4chain) [reply config/flags.go:173]: I'm not sure 😄

**Review by arkadiuszos4chain — APPROVED** [review]

**Pull request merged**